ast/compile_test.go

```diff
@@ -1515,6 +1515,118 @@ func TestCompilerRewriteExprTerms(t *testing.T) {
 	}
 }
 
+func TestIllegalFunctionCallRewrite(t *testing.T) {
+	cases := []struct {
+		note           string
+		module         string
+		expectedErrors []string
+	}{
+		/*{
+					note: "function call override in function value",
```

Comment:
Should this commented out code be here?

Reply:
No. This will be removed. I kept it in for reviewing purposes, to illustrate the limitations of the current solution, as this test case will fail.